pkg/controllermanager/controller/shoot/shoot_maintenance_control.go

```diff
@@ -172,10 +172,11 @@ func (c *defaultMaintenanceControl) Maintain(shootObj *gardencorev1beta1.Shoot,
 			return nil, fmt.Errorf("auto update section of Shoot %s/%s changed mid-air", s.Namespace, s.Name)
 		}
 
-		delete(s.Annotations, v1beta1constants.GardenerOperation)
```

Comment:
Looks good, the only thing : why is the operations not removed any more?

Reply:
It doesn't make sense to explicitly remove it because, in the next line, we add it again with the "reconcile" value. Hence, we can overwrite it directly and get the same result.